python/tskit/trees.py

```diff
@@ -171,17 +171,17 @@ class Edge(SimpleContainer):
         To obtain further information about a node with a given ID, use
         :meth:`.TreeSequence.node`.
     :vartype child: int
+    :ivar id: The integer ID of this edge. Varies from 0 to
+        :attr:`.TreeSequence.num_edges` - 1.
+    :vartype id: int
     """
-    def __init__(self, left, right, parent, child):
+    def __init__(self, left, right, parent, child, id_=None):
+        self.id = id_
         self.left = left
         self.right = right
         self.parent = parent
         self.child = child
 
-    def __repr__(self):
```

Comment:
Why delete this?

Reply:
The other SimpleContainer classes don't have an explicit __repr__ method (the rely on the base class). Having one here meant that we ended up not printing the id on display. So this seems redundant to me

Reply:
I guess it's probably unncessary. I put it in to make reading the edges easier (note the precision for left and right)

Reply:
Oh, good point about the precision. I'll reinstate with the id var added, then.

Reply:
Done